File: backend/app/schemas/kakao_auth.py
"""카카오 OAuth 이후 앱 회원가입 API 스키마."""

from typing import Literal

from pydantic import BaseModel, Field

RegistrationStatus = Literal["authenticated", "signup_required"]


class RegistrationResponse(BaseModel):
    status: RegistrationStatus


class AuthUserInfo(BaseModel):
    id: str
    email: str | None = None
    nickname: str | None = None
    profile_image: str | None = None


class KakaoAuthResponse(BaseModel):
    status: RegistrationStatus
    user: AuthUserInfo


class KakaoSignUpRequest(BaseModel):
    nickname: str | None = Field(default=None, max_length=20)
    agree_terms: Literal[True]
    agree_privacy: Literal[True]
    agree_marketing: bool = False

File: frontend/src/api/kakaoAuth.ts
import { apiDelete, apiGet, apiPost } from './client'

export type RegistrationStatus = 'authenticated' | 'signup_required'

export interface KakaoUser {
  id: string
  email: string | null
  nickname: string | null
  profile_image: string | null
}

export interface KakaoAuthResult {
  status: RegistrationStatus
  user: KakaoUser
}

export interface KakaoSignUpRequest {
  nickname: string | null
  agree_terms: true
  agree_privacy: true
  agree_marketing: boolean
}

export function kakaoLogin(): Promise<KakaoAuthResult> {
  return apiPost<KakaoAuthResult>('/api/v1/auth/kakao/login', {})
}

export function getRegistration(): Promise<{ status: RegistrationStatus }> {
  return apiGet<{ status: RegistrationStatus }>('/api/v1/auth/registration')
}

export function abandonKakaoSignup(): Promise<{ deleted: boolean }> {
  return apiDelete<{ deleted: boolean }>('/api/v1/auth/kakao/pending')
}

export function completeKakaoSignup(body: KakaoSignUpRequest): Promise<KakaoAuthResult> {
  return apiPost<KakaoAuthResult>('/api/v1/auth/signup', body)
}
